pkg/features/list.go

```diff
@@ -4,51 +4,60 @@ package features
-	NetworkGraphPatternFly = registerFeature("Enable PatternFly version of NetworkGraph", "ROX_NETWORK_GRAPH_PATTERNFLY", true)
+	NetworkGraphPatternFly = registerUnchangeableFeature("Enable PatternFly version of NetworkGraph", "ROX_NETWORK_GRAPH_PATTERNFLY", true)
 
 	// ClairV4Scanner enables Clair v4 as an Image Integration option
-	ClairV4Scanner = registerFeature("Enable Clair v4 as an Image Integration option", "ROX_CLAIR_V4_SCANNING", true)
+	ClairV4Scanner = registerUnchangeableFeature("Enable Clair v4 as an Image Integration option", "ROX_CLAIR_V4_SCANNING", true)
 
 	// RoxSyslogExtraFields enables user to add additional key value pairs in syslog alert notification in cef format.
-	RoxSyslogExtraFields = registerFeature("Enable extra fields for syslog integration", "ROX_SYSLOG_EXTRA_FIELDS", true)
+	RoxSyslogExtraFields = registerUnchangeableFeature("Enable extra fields for syslog integration", "ROX_SYSLOG_EXTRA_FIELDS", true)
 
 	// SourcedAutogeneratedIntegrations enables adding a "source" to autogenerated integrations.
-	SourcedAutogeneratedIntegrations = registerFeature("Enable autogenerated integrations with cluster/namespace/secret source", "ROX_SOURCED_AUTOGENERATED_INTEGRATIONS", false)
+	SourcedAutogeneratedIntegrations = registerUnchangeableFeature("Enable autogenerated integrations with cluster/namespace/secret source", "ROX_SOURCED_AUTOGENERATED_INTEGRATIONS", false)
 
 	// VulnMgmtWorkloadCVEs enables APIs and UI pages for the VM Workload CVE enhancements
-	VulnMgmtWorkloadCVEs = registerFeature("Vuln Mgmt Workload CVEs", "ROX_VULN_MGMT_WORKLOAD_CVES", false)
+	VulnMgmtWorkloadCVEs = registerUnchangeableFeature("Vuln Mgmt Workload CVEs", "ROX_VULN_MGMT_WORKLOAD_CVES", false)
 
 	// PostgresBlobStore enables the creation of the Postgres Blob Store
-	PostgresBlobStore = registerFeature("Postgres Blob Store", "ROX_POSTGRES_BLOB_STORE", false)
+	PostgresBlobStore = registerUnchangeableFeature("Postgres Blob Store", "ROX_POSTGRES_BLOB_STORE", false)
 
 	// VulnMgmtReportingEnhancements enables APIs and UI pages for VM Reporting enhancements including downloadable reports
-	VulnMgmtReportingEnhancements = registerFeature("Vuln Mgmt Reporting Enhancements", "ROX_VULN_MGMT_REPORTING_ENHANCEMENTS", false)
+	VulnMgmtReportingEnhancements = registerUnchangeableFeature("Vuln Mgmt Reporting Enhancements", "ROX_VULN_MGMT_REPORTING_ENHANCEMENTS", false)
 
 	// StoreEventHashes stores the hashes of successfully processed objects we receive from Sensor into the database
-	StoreEventHashes = registerFeature("Store Event Hashes", "ROX_STORE_EVENT_HASHES", false)
+	StoreEventHashes = registerUnchangeableFeature("Store Event Hashes", "ROX_STORE_EVENT_HASHES", false)
+
+	// PostgresDatastoreEnabled toggles whether to use Postgres for the datastore or not.
+	PostgresDatastoreEnabled = registerUnchangeableFeature("Enable PostgresSQL Datastore", "ROX_POSTGRES_DATASTORE", true)
```

Comment:
Question: this was one of the three "hacked" feature flags, that were changeable through an env var before, so why have you made it `unchangeable` in this example?

Reply:
Because it was changed to being "unchangeable" in the env vars as well: https://github.com/stackrox/stackrox/commit/da708eb2d53a7a0c17ef8781169efb760afda49e